s/core.go

```diff
@@ -162,7 +163,7 @@ func (spec SupervisorSpec) start(parentCtx context.Context, parentName string) (
 	startCh := make(chan error)
 
 	// terminateCh is used when waiting for cancelFn to complete
-	terminateCh := make(chan struct{})
+	terminateCh := make(chan error)
```

Comment:
Here too, I might consider an alias or type? Don't know for sure.

Reply:
```go
type terminationError = error
```
???